### PR TITLE
Analyzer variable qualification

### PR DESCRIPTION
**Pull request by Claes65**

### CODE REVIEW

File: libs/natlint/src/main/java/org/amshove/natlint/analyzers/QualifiedVariableAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	{
+		context.registerNodeAnalyzer(IVariableReferenceNode.class, this::analyzeQualifiedVariable);
+		context.registerNodeAnalyzer(ITypedVariableNode.class, this::analyzeLevel1Variable);
+	}
+
+	@Override
+	public void beforeAnalyzing(IAnalyzeContext context)
+	{
+		isQualifiedVarsOff = !context.getConfiguration(context.getModule().file(), "natls.style.qualifyvars", OPTION_FALSE).equalsIgnoreCase(OPTION_TRUE);
+		isLevel1VarsOff = !context.getConfiguration(context.getModule().file(), "natls.style.level1vars", OPTION_FALSE).equalsIgnoreCase(OPTION_TRUE);
```
Comment (MarkusAmshove) [inline libs/natlint/src/main/java/org/amshove/natlint/analyzers/QualifiedVariableAnalyzer.java:48]: Lets call that option `natls.style.disallowtoplevelvars`. Makes the attempt a bit clearer :)

File: libs/natlint/src/main/java/org/amshove/natlint/analyzers/QualifiedVariableAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+		if (!NodeUtil.moduleContainsNode(context.getModule(), node))
+		{
+			return;
+		}
+
+		var variable = (IVariableReferenceNode) node;
+		if (variable.reference()instanceof ITypedVariableNode typedVariable && !typedVariable.qualifiedName().equals(variable.referencingToken().symbolName()))
+		{
+			context.report(VARIABLE_SHOULD_BE_QUALIFIED.createFormattedDiagnostic(variable.referencingToken(), variable.referencingToken().source()));
```
Comment (MarkusAmshove) [inline libs/natlint/src/main/java/org/amshove/natlint/analyzers/QualifiedVariableAnalyzer.java:71]: You can use `createDiagnostic(variable.referencingToken())`.

`createFormattedDiagnostic()` is used when you have a placeholder in your `DiagnosticDescription`.

If you want to have the variable name included, you can change the message template of the description from
`"Variable should be qualified"`
to
`"Variable %s should be qualified"` and keep the `createFormattedDiagnostic()`.

If you want to go further, you can change it to
`"Variable %s should be qualified, change it to %s"` and call `createFormattedDiagnostic(variable.referencingToken(), variable.referencingToken().source(), typedVariable.qualifiedName()` but that might be too much :)

**Review by MarkusAmshove — CHANGES_REQUESTED** [review]
Some small nitpicks.

You should update your branch with current `main`, because the diagnostic IDs will be off, I added a new analyzer in the meantime.
I suggest to change your ids to NL018 and NL019 before merging `main`, because the `NL017` file might raise merge conflicts if you don't do the change beforehand :/

File: libs/natlint/src/main/java/org/amshove/natlint/analyzers/QualifiedVariableAnalyzer.java (hunk 2 above)
Comment (Claes65) [reply libs/natlint/src/main/java/org/amshove/natlint/analyzers/QualifiedVariableAnalyzer.java:71]: I like the latter :)

**Review by MarkusAmshove — APPROVED** [review]

**Pull request merged**